src/java/org/apache/cassandra/db/guardrails/GuardrailsMBean.java

```diff
@@ -468,6 +468,30 @@
      */
     void setWriteConsistencyLevelsDisallowedCSV(String consistencyLevels);
 
+    /**
+     * @return The threshold to warn when encountering partitions larger than threshold, as a string  formatted as in,
+     * for example, {@code 10GiB}, {@code 20MiB}, {@code 30KiB} or {@code 40B}. A {@code null} value means disabled.
+     */
+    @Nullable
+    String getPartitionSizeWarnThreshold();
+
+    /**
+     * @return The threshold to fail when encountering partitions larger than threshold, as a string  formatted as in,
```

Comment:
I would reword all the 'fail' variant comments to highlight the operation won't fail and it will only get logged at ERROR level. Otherwise it's confusing imo.

Reply:
I guess the meaning of "fail" on all background guardrails like this one is logging an error and emit a failure event. Not very sure how to rename, but I have added a line telling that it will not throw an exception to avoid stopping the write.

Reply:
+1